Replacing old color variables

robinrendle opened the pull request:
Final refactoring, removes all old color variables.

Next step is to make common color vars for text/heading colors and backgrounds.

Commit:
fix(ui): refactored old color vars

Commit:
tests

Commit:
fix(ui): remove old color variables

Commit:
tests

Commit:
replaced old vars

Commit:
removed final color vars

Commit:
tests

Commit:
replaced red vars

Commit:
tests

Commit:
fixed pink color var

robinrendle committed:
tests

robinrendle committed:
added borderDark/borderLight vars

robinrendle committed:
tests

billyvg commented:
@robinrendle can we undo the border changes here as well?

robinrendle committed:
fixed border color variables

robinrendle committed:
docs border vars

robinrendle committed:
replaced purpleDarkest variable

robinrendle commented:
@billyvg Done and done. Ready to go now I think!

billyvg commented on src/sentry/static/sentry/app/components/events/interfaces/breadcrumbs/styles.tsx:
```diff
@@ -19,7 +19,7 @@ const BreadCrumb = styled('li')<{hasError?: boolean}>`
     bottom: 0;
     left: 32px;
-    background: ${p => p.theme.borderLight};
+    background: ${p => p.theme.gray300};
```
```suggestion
    background: ${p => p.theme.borderLight};
```

billyvg commented on src/sentry/static/sentry/app/components/events/interfaces/openInContextLine.tsx:
```diff
@@ -60,7 +60,7 @@ const OpenInContainer = styled('div')<{columnQuantity: number}>`
   grid-template-columns: repeat(${p => p.columnQuantity}, max-content);
   grid-gap: ${space(1)};
-  color: ${p => p.theme.purple2};
+  color: ${p => p.theme.gray600};
```
I assume this is intentional

robinrendle replied:
Oddly enough, yep it’s intentional.

billyvg commented on src/sentry/static/sentry/app/components/events/interfaces/spans/spanBar.tsx:
```diff
@@ -970,16 +972,16 @@ const getTogglerTheme = ({
 
   if (disabled) {
     return `
-    background: ${buttonTheme.background};
```
This feels like a similar situation as border, but it's probably not used as much so this might be ok

robinrendle replied:
👍

billyvg commented on src/sentry/static/sentry/app/components/projects/bookmarkStar.tsx:
```diff
@@ -75,7 +75,7 @@ BookmarkStar.propTypes = {
 
 const Star = styled(InlineSvg)<{isBookmarked: boolean}>`
-  color: ${p => (p.isBookmarked ? p.theme.orange300 : p.theme.gray400)};
+  color: ${p => (p.isBookmarked ? p.theme.orange300 : p.theme.borderDark)};
```
Intended?

billyvg commented on src/sentry/static/sentry/app/components/panels/hintPanelItem.tsx:
```diff
@@ -5,8 +5,8 @@ import space from 'app/styles/space';
   padding: ${space(2)};
-  border-top: 1px solid ${p => p.theme.borderLighter};
-  border-left: 1px solid ${p => p.theme.borderLighter};
+  border-top: 1px solid ${p => p.theme.gray100};
```
```suggestion
  border-top: 1px solid ${p => p.theme.borderLighter};
```

billyvg commented on src/sentry/static/sentry/app/components/panels/hintPanelItem.tsx:
```diff
@@ -5,8 +5,8 @@ import space from 'app/styles/space';
-  border-top: 1px solid ${p => p.theme.borderLighter};
-  border-left: 1px solid ${p => p.theme.borderLighter};
+  border-top: 1px solid ${p => p.theme.gray100};
+  border-left: 1px solid ${p => p.theme.gray100};
```
```suggestion
  border-left: 1px solid ${p => p.theme.borderLighter};
```

billyvg commented on src/sentry/static/sentry/app/components/charts/worldMapChart.jsx:
```diff
@@ -58,7 +58,7 @@ export default class WorldMapChart extends React.Component {
           normal: {
             areaColor: theme.gray400,
-            borderColor: theme.borderLighter,
+            borderColor: theme.gray100,
```
```suggestion
            borderColor: theme.borderLighter,
```

billyvg commented on src/sentry/static/sentry/app/components/charts/worldMapChart.jsx:
```diff
@@ -81,13 +81,13 @@ export default class WorldMapChart extends React.Component {
       <BaseChart
         options={{
-          backgroundColor: theme.borderLighter,
+          backgroundColor: theme.gray100,
```
```suggestion
          backgroundColor: theme.borderLighter,
```

billyvg reviewed (commented):
@robinrendle We may want to consider a theme key that maintains the constraint between border color and another property (e.g. background color). E.g. if we decide to change border color tomorrow, it'd be a bit more tedious to find all of the colors or background-colors that are intended to match the border color.

Similar idea with backgroundColor being replaced with white

robinrendle committed:
replaced borderDark

robinrendle committed:
fixed border colors

robinrendle committed:
fixed styles

robinrendle committed:
fixed regression

robinrendle commented:
@billyvg Thanks for catching all my dumbness. Fixed the visual regressions and added back `borderDark`, `borderLight`, and `borderLighter` vars for now. I also agree about having more semantic variable names.

robinrendle committed:
fixed span bg color regression

robinrendle committed:
fixed tests

robinrendle commented:
Testing to make sure getsentry is okay with these changes: https://github.com/getsentry/getsentry/pull/4096

billyvg reviewed (approved):
🚀
